scripts/examples/esp_example.sh

```diff
@@ -30,20 +30,40 @@ if [ -z "$app" ]; then
     exit 1
 fi
 
-source "$IDF_PATH/export.sh"
+if [ "$sdkconfig_name" == "sdkconfig.defaults.esp32h2" ]; then
+    pushd .
+    cd "$IDF_PATH"
+    git fetch origin master
+    git checkout 10f3aba770
```

Comment:
This seems off - IDF should be configured in advance. This checks out a specific version and it will not revert it.

I believe build scripts should not touch SDKs but rather  check that SDKs are correctpy configured.
We need an image that has the right version.

Reply:
The reason I care about this is that developers use a single image to compile things. I want to be able to compile anything without making my SDK/tree dirty. I should be able to build H2 as well as others after each other in the same environment.

CI will start fresh every time, so we do not care as much, however developer images (usually the chip-build-vscode and similar variants) are persistent.

Reply:
We could add a separate docker image in that case or make this update part of CI explicitly rather than build scripts. Build scripts should not touch SDK/source in a such strong way.